web/frontend/src/components/do-command/index.svelte

```diff
@@ -1,87 +1,104 @@
 <script lang="ts">
-
 import { commonApi } from '@viamrobotics/sdk';
 import { notify } from '@viamrobotics/prime';
 import { resourceNameToString } from '@/lib/resource';
 import { doCommand } from '@/api/do-command';
 import Collapse from '@/lib/components/collapse.svelte';
 import { useRobotClient } from '@/hooks/robot-client';
+import { getClientByType } from './get-client-by-type';
+import { Button, Label, SearchableSelect } from '@viamrobotics/prime-core';
+import { isUnimplementedError } from './errors';
 
 export let resources: commonApi.ResourceName.AsObject[];
 
 const { robotClient } = useRobotClient();
 
-let selectedComponent = '';
-let input = '';
+let selectedComponent = undefined as
+  | commonApi.ResourceName.AsObject
+  | undefined;
+let input = '{}';
 let output = '';
 let executing = false;
 
-const handleDoCommand = async (name: string, command: string) => {
-  if (!name || !command) {
+const handleDoCommand = async (type: string, name: string, command: string) => {
```

Comment:
Same question: why aren't we typing `type` more strictly?

Reply:
Because we are basing the type off of the `ResourceName.AsObject.subtype` property which is defined as a `string | undefined`. We cannot use a string to index the `RobotClient` to get the various services, so I need a way to say for a fact "take any generic string, guarantee it is a service key available on the robot client, then return that client if it exists". So ultimately we will have to cast that `subtype` string to something we can say for definite is available. Though I was rethinking it a bit and pushed up an update that removed a lot of the redundant string typing.